### PR TITLE
releases: add 4.0.0 GA release notes

### PR DESCRIPTION
**Pull request by toutdesuite**
### What is changed, added or deleted? (Required)

add 4.0.0 GA release notes

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### CODE REVIEW

File: releases/release-4.0.0-ga.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+* TiDB
+    + 新增 `committer-concurrency` 配置项，用于控制 retry commit 阶段的 goroutine 数量 [#16849](https://github.com/pingcap/tidb/pull/16849)
+    + 支持 `show table partition regions` 语法 [#17294](https://github.com/pingcap/tidb/pull/17294)
+    + 新增 `tmp-storage-quota` 配置项用于限制 tidb-server 使用的临时磁盘空间 [#15700](https://github.com/pingcap/tidb/pull/15700)
+    + 创建和更改表时新增检查分区表是否使用唯一前缀索引的功能 [#17213](https://github.com/pingcap/tidb/pull/17213)
+    + 支持 `insert/replace into tbl_name partition`(`partition_name_list`) 语句 [#17313](https://github.com/pingcap/tidb/pull/17313)
+    + Distinct 函数支持检查 collations 的值 [#17240](https://github.com/pingcap/tidb/pull/17240)
+    + 哈希分区裁剪时支持 `is null` 过滤条件 [#17310](https://github.com/pingcap/tidb/pull/17310)
+    + 分区表中支持 `admin check index`、`admin cleanup index` 和 `admin recover index` [#17392](https://github.com/pingcap/tidb/pull/17392) [#17405](https://github.com/pingcap/tidb/pull/17405)  [#17317](https://github.com/pingcap/tidb/pull/17317)
```
Comment (lilin90) [inline releases/release-4.0.0-ga.md:41]:
```suggestion
    + 分区表中支持 `admin check index`、`admin cleanup index` 和 `admin recover index` [#17392](https://github.com/pingcap/tidb/pull/17392) [#17405](https://github.com/pingcap/tidb/pull/17405) [#17317](https://github.com/pingcap/tidb/pull/17317)
```

File: releases/release-4.0.0-ga.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+* TiKV
+    + 支持 `ascii_bin` 和 `latin1_bin` 编码的排序规则 [#7919](https://github.com/tikv/tikv/pull/7919)
+
+* PD
+    + 支持为内置的 TiDB Dashboard 指定反向代理资源前缀 [#2457](https://github.com/pingcap/pd/pull/2457)
+    + PD client Region 相关接口支持返回 `pending peer` 和 `down peer` 的信息 [#2443](https://github.com/pingcap/pd/pull/2443)
+    + 添加 `Direction of hotspot move leader`、`Direction of hotspot move peer` 和 `Hot cache read entry number` 等监控 [#2448](https://github.com/pingcap/pd/pull/2448)
+
+* Tools
+    + BR
```
Comment (lilin90) [inline releases/release-4.0.0-ga.md:58]:
```suggestion
    + Backup & Restore (BR)
```

File: releases/release-4.0.0-ga.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+* TiCDC
+    + 优化 Changefeed 配置文件的结构，提升易用性 [#588](https://github.com/pingcap/ticdc/pull/588)
+    + 新增 `ignore-txn-start-ts` 配置项，过滤事务时条件由原来的 `commit_ts` 改为 `start_ts` [#589](https://github.com/pingcap/ticdc/pull/589)
+
+## 重点修复的 Bug
+
+* TiKV
+    + 修复 BR 备份时出现 `DefaultNotFound` 错误的问题 [#7937](https://github.com/tikv/tikv/pull/7937)
+    + 修复 ReadIndex 因响应的数据包乱序时导致系统 panic 问题 [#7930](https://github.com/tikv/tikv/pull/7930)
+    + 修复 TiKV 重启后由于 snapshot 文件被错误的删除导致系统 panic 问题 [#7927](https://github.com/tikv/tikv/pull/7927)
```
Comment (ran-huang) [inline releases/release-4.0.0-ga.md:26]:
```suggestion
    + 修复 TiKV 重启后由于 snapshot 文件被错误地删除导致系统 panic 问题 [#7927](https://github.com/tikv/tikv/pull/7927)
```

File: releases/release-4.0.0-ga.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+* TiFlash
+    + 修复因 Raft Admin Command 处理逻辑不正确，系统 panic 导致可能会出现丢数据问题
```
Comment (ran-huang) [inline releases/release-4.0.0-ga.md:29]:
```suggestion
    + 修复因 Raft Admin Command 处理逻辑不正确，系统 panic 导致可能会出现丢数据的问题
```

File: releases/release-4.0.0-ga.md (hunk 1 above)
Comment (ran-huang) [inline releases/release-4.0.0-ga.md:41]:
```suggestion
    + 分区表中支持 `admin check index`、`admin cleanup index` 和 `admin recover index` [#17392](https://github.com/pingcap/tidb/pull/17392) [#17405](https://github.com/pingcap/tidb/pull/17405) [#17317](https://github.com/pingcap/tidb/pull/17317)
```

File: releases/release-4.0.0-ga.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    + 修复收发消息有不符合预期的超时，禁止合并发向 TiFlash 的消息 [#17307](https://github.com/pingcap/tidb/pull/17307)
+    + 修复分区裁剪时未正确区分有符号/无符号整数的错误，并提高了性能 [#17230](https://github.com/pingcap/tidb/pull/17230)
+    + 修复 3.1.1 升级到 4.0 时由于 `mysql.user` 表不兼容导致升级失败的问题 [#17300](https://github.com/pingcap/tidb/pull/17300)
+    + 修复 `update` 语句分区选择不正确的问题 [#17305](https://github.com/pingcap/tidb/pull/17305)
+    + 修复从 TiKV 收到未知错误消息时系统 panic 的问题 [#17380](https://github.com/pingcap/tidb/pull/17380)
+    + 修复创建按 key 分区的分区表由于处理逻辑不正确导致系统 panic 的问题 [#17242](https://github.com/pingcap/tidb/pull/17242)
+    + 修复某些情况下优化器处理逻辑不正确导致错误地选择 `Index Merge Join` 的问题 [#17365](https://github.com/pingcap/tidb/pull/17365)
+    + 修复 Grafana 中 `SELECT` 语句的 `duration` 的监控指标不准确的问题 [#16561](https://github.com/pingcap/tidb/pull/16561)
+    + 修复当系统发生错误时，GC 线程卡住的问题 [#16915](https://github.com/pingcap/tidb/pull/16915)
+    + 修复 列的类型是 `Boolean` 时 由于 `UNIQUE` 约束比较的逻辑不正确导致输出结果不正确的问题 [#17306](https://github.com/pingcap/tidb/pull/17306)
```
Comment (ran-huang) [inline releases/release-4.0.0-ga.md:78]:
```suggestion
    + 修复当列的类型是 `Boolean` 时，由于 `UNIQUE` 约束比较的逻辑不正确导致输出结果不正确的问题 [#17306](https://github.com/pingcap/tidb/pull/17306)
```

File: releases/release-4.0.0-ga.md [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+    + 修复 Search log 功能正则表达式匹配的行为与其他组件不一致的问题
+    + 默认关闭 Raft Compact Log Command 延迟处理的优化，避免节点大量写入数据时重启时间过长的问题
+    + 修复部分场景因 TiDB 执行 `DROP DATABASE` 处理逻辑不正确导致系统启动不成功的问题
+    + 修复采集 `Server_info` 中 CPU 信息方式与其他组件不一样的问题
+    + 修复开启 `batch coprocessor` 功能，执行 `Query` 报错 `Too Many Pings` 的问题
+    + 修复 Dashboard 因未上报 `deploy path` 信息导致相关信息显示不正确的问题
+
+* TiKV
+    + 修复 BR 备份时出现 `DefaultNotFound` 错误的问题 [#7937](https://github.com/tikv/tikv/pull/7937)
+    + 修复 ReadIndex 因响应的数据包乱序时导致系统 panic 问题 [#7930](https://github.com/tikv/tikv/pull/7930)
```
Comment (lilin90) [inline releases/release-4.0.0-ga.md:95]:
```suggestion
    + 修复 `ReadIndex` 因响应的数据包乱序时导致系统 panic 问题 [#7930](https://github.com/tikv/tikv/pull/7930)
```

Comment (ran-huang) [inline releases/release-4.0.0-ga.md:95]: @scsldb 这几条已经在**重点修复的 Bug**里写过了，请确认是否需要重复写在这里？

File: releases/release-4.0.0-ga.md [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+
+* PD
+    + 修复使用 pd-ctl 删除 `evict-leader-scheduler` 或者 `grant-leader-scheduler` 时报 404 错误的问题 [#2446](https://github.com/pingcap/pd/pull/2446)
+    + 修复当存在 TiFlash 副本的时候，可能会导致 `presplit` 功能无法正常使用的问题 [#2447](https://github.com/pingcap/pd/pull/2447)
+
+* Tools
+    * BR
+        + 修复从云存储恢复数据时因网络原因导致恢复失败的问题 [#298](https://github.com/pingcap/br/pull/298)
+    * TiCDC
+        + 修复若干因数据争用（data race）导致系统 panic 的问题 [#565](https://github.com/pingcap/ticdc/pull/565) [#566](https://github.com/pingcap/ticdc/pull/566)
```
Comment (ran-huang) [inline releases/release-4.0.0-ga.md:109]:
```suggestion
        + 修复若干因数据争用 (data race) 导致系统 panic 的问题 [#565](https://github.com/pingcap/ticdc/pull/565) [#566](https://github.com/pingcap/ticdc/pull/566)
```

File: releases/release-4.0.0-ga.md [hunk 8]
```diff
@@ -0,0 +1,111 @@
+---
+title: TiDB 4.0.0 GA Release Notes
+category: Releases
+---
+
+# TiDB 4.0.0 GA Release Notes
+
+发版日期：2020 年 5 月 28 日
+
+TiDB 版本：4.0.0 GA
```
Comment (ran-huang) [inline releases/release-4.0.0-ga.md:10]: @scsldb 这里是否要带 `GA`？3.0 GA 的 rn 里只写了 `3.0.0`.
https://docs.pingcap.com/zh/tidb/dev/release-3.0-ga

Comment (lilin90) [reply releases/release-4.0.0-ga.md:10]: 去掉 GA 吧，跟 repo 里实际的版本号一致：https://github.com/pingcap/tidb/releases/tag/v4.0.0

Comment (lilin90) [inline releases/release-4.0.0-ga.md:10]:
```suggestion
TiDB 版本：4.0.0
```

File: releases/release-4.0.0-ga.md [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+    + 修复分区裁剪时未正确区分有符号/无符号整数的错误，并提高了性能 [#17230](https://github.com/pingcap/tidb/pull/17230)
+    + 修复 3.1.1 升级到 4.0 时由于 `mysql.user` 表不兼容导致升级失败的问题 [#17300](https://github.com/pingcap/tidb/pull/17300)
+    + 修复 `update` 语句分区选择不正确的问题 [#17305](https://github.com/pingcap/tidb/pull/17305)
+    + 修复从 TiKV 收到未知错误消息时系统 panic 的问题 [#17380](https://github.com/pingcap/tidb/pull/17380)
+    + 修复创建按 key 分区的分区表由于处理逻辑不正确导致系统 panic 的问题 [#17242](https://github.com/pingcap/tidb/pull/17242)
+    + 修复某些情况下优化器处理逻辑不正确导致错误地选择 `Index Merge Join` 的问题 [#17365](https://github.com/pingcap/tidb/pull/17365)
+    + 修复 Grafana 中 `SELECT` 语句的 `duration` 的监控指标不准确的问题 [#16561](https://github.com/pingcap/tidb/pull/16561)
+    + 修复当系统发生错误时，GC 线程卡住的问题 [#16915](https://github.com/pingcap/tidb/pull/16915)
+    + 修复 列的类型是 `Boolean` 时 由于 `UNIQUE` 约束比较的逻辑不正确导致输出结果不正确的问题 [#17306](https://github.com/pingcap/tidb/pull/17306)
+    + 修复 `tidb_opt_agg_push_down` 开启且聚合函数下推分区表信息时由于逻辑处理不正确导致系统 panic 的问题 [#17328](https://github.com/pingcap/tidb/pull/17328)
```
Comment (lilin90) [inline releases/release-4.0.0-ga.md:79]:
```suggestion
    + 修复 `tidb_opt_agg_push_down` 开启且聚合函数下推分区表信息时，由于逻辑处理不正确导致系统 panic 的问题 [#17328](https://github.com/pingcap/tidb/pull/17328)
```

File: releases/release-4.0.0-ga.md [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+    + 修复 `tidb.toml` 中的配置项 `isolation-read` 不生效时的问题 [#17322](https://github.com/pingcap/tidb/pull/17322)
+    + 修复通过 `hint` 强制执行流式聚合时由于逻辑处理不正确导致输出的结果顺序不正确的问题 [#17347](https://github.com/pingcap/tidb/pull/17347)
+    + 修复不同 `SQL_MODE` 时 `insert` 处理 `DIV` 的行为 [#17314](https://github.com/pingcap/tidb/pull/17314)
+
+* TiFlash
+    + 修复 Search log 功能正则表达式匹配的行为与其他组件不一致的问题
+    + 默认关闭 Raft Compact Log Command 延迟处理的优化，避免节点大量写入数据时重启时间过长的问题
+    + 修复部分场景因 TiDB 执行 `DROP DATABASE` 处理逻辑不正确导致系统启动不成功的问题
+    + 修复采集 `Server_info` 中 CPU 信息方式与其他组件不一样的问题
+    + 修复开启 `batch coprocessor` 功能，执行 `Query` 报错 `Too Many Pings` 的问题
```
Comment (lilin90) [inline releases/release-4.0.0-ga.md:90]:
```suggestion
    + 修复开启 `batch coprocessor` 功能时，执行 `Query` 报错 `Too Many Pings` 的问题
```

File: releases/release-4.0.0-ga.md [hunk 11]
```diff
@@ -0,0 +1,111 @@
...
+    + 修复 Dashboard 因未上报 `deploy path` 信息导致相关信息显示不正确的问题
+
+* TiKV
+    + 修复 BR 备份时出现 `DefaultNotFound` 错误的问题 [#7937](https://github.com/tikv/tikv/pull/7937)
+    + 修复 ReadIndex 因响应的数据包乱序时导致系统 panic 问题 [#7930](https://github.com/tikv/tikv/pull/7930)
+    + 修复读请求回调函数没有被调用，导致返回非预期错误的问题 [#7921](https://github.com/tikv/tikv/pull/7921)
+    + 修复 TiKV 重启后由于 snapshot 文件被错误的删除导致系统 panic 问题 [#7927](https://github.com/tikv/tikv/pull/7927)
```
Comment (lilin90) [inline releases/release-4.0.0-ga.md:97]:
```suggestion
    + 修复 TiKV 重启后由于 snapshot 文件被错误删除导致系统 panic 问题 [#7927](https://github.com/tikv/tikv/pull/7927)
```

File: releases/release-4.0.0-ga.md [hunk 12]
```diff
@@ -0,0 +1,111 @@
...
+    + 修复使用 pd-ctl 删除 `evict-leader-scheduler` 或者 `grant-leader-scheduler` 时报 404 错误的问题 [#2446](https://github.com/pingcap/pd/pull/2446)
+    + 修复当存在 TiFlash 副本的时候，可能会导致 `presplit` 功能无法正常使用的问题 [#2447](https://github.com/pingcap/pd/pull/2447)
+
+* Tools
+    * BR
+        + 修复从云存储恢复数据时因网络原因导致恢复失败的问题 [#298](https://github.com/pingcap/br/pull/298)
+    * TiCDC
+        + 修复若干因数据争用（data race）导致系统 panic 的问题 [#565](https://github.com/pingcap/ticdc/pull/565) [#566](https://github.com/pingcap/ticdc/pull/566)
+        + 修复若干因处理逻辑不正确导致资源泄露或系统阻塞的问题 [#574](https://github.com/pingcap/ticdc/pull/574) [#586](https://github.com/pingcap/ticdc/pull/586)
+        + 修复 CLI 因连接不上 PD 时会导致命令行阻塞的问题 [#579](https://github.com/pingcap/ticdc/pull/579)
```
Comment (lilin90) [inline releases/release-4.0.0-ga.md:111]:
```suggestion
        + 修复 CLI 因连接不上 PD 导致命令行阻塞的问题 [#579](https://github.com/pingcap/ticdc/pull/579)
```

**Comment by lilin90** [comment]
@toutdesuite Please address the comments.

File: TOC.md [hunk 13]
```diff
@@ -456,6 +456,7 @@
 + 版本发布历史
   + [发布版本汇总](/releases/release-notes.md)
   + v4.0
+    - [4.0.0 GA](/releases/release-4.0.0-ga.md)
```
Comment (lilin90) [inline TOC.md:459]:
```suggestion
    - [4.0 GA](/releases/release-4.0-ga.md)
```

File: releases/release-4.0.0-ga.md [hunk 14]
```diff
@@ -0,0 +1,111 @@
+---
```
Comment (lilin90) [inline releases/release-4.0.0-ga.md:1]: Please update the file name.

File: releases/release-notes.md [hunk 15]
```diff
@@ -10,6 +10,7 @@ TiDB 历史版本发布声明如下：
 
 ## 4.0
 
+- [4.0.0 GA](/releases/release-4.0.0-ga.md)
```
Comment (lilin90) [inline releases/release-notes.md:13]:
```suggestion
- [4.0 GA](/releases/release-4.0-ga.md)
```

**Comment by toutdesuite** [comment]
@lilin90 Comments addressed

File: releases/release-4.0-ga.md [hunk 16]
```diff
@@ -0,0 +1,111 @@
+---
+title: TiDB 4.0.0 GA Release Notes
```
Comment (lilin90) [inline releases/release-4.0-ga.md:2]:
```suggestion
title: TiDB 4.0 GA Release Notes
```

File: releases/release-4.0-ga.md [hunk 17]
```diff
@@ -0,0 +1,111 @@
...
+category: Releases
+---
+
+# TiDB 4.0.0 GA Release Notes
```
Comment (lilin90) [inline releases/release-4.0-ga.md:6]:
```suggestion
# TiDB 4.0 GA Release Notes
```

File: releases/release-4.0.0-ga.md (hunk 6 above)
Comment (lilin90) [reply releases/release-4.0.0-ga.md]: 先留着吧

**Pull request merged**